File: backend/src/main/java/com/minidocto/model/Review.java
package com.minidocto.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "reviews")
public class Review {
    @Id
    private String id;
    
    private String professionalId;
    private String patientId;
    private String appointmentId;
    private Integer rating;
    private String comment;
    private LocalDateTime createdAt;
    
    public Review() {
        this.createdAt = LocalDateTime.now();
    }
}

File: patient_app/lib/models/review.dart
class Review {
  final String? id;
  final String professionalId;
  final String patientId;
  final String appointmentId;
  final int rating;
  final String? comment;
  final DateTime createdAt;

  Review({
    this.id,
    required this.professionalId,
    required this.patientId,
    required this.appointmentId,
    required this.rating,
    this.comment,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      professionalId: json['professionalId'],
      patientId: json['patientId'],
      appointmentId: json['appointmentId'],
      rating: json['rating'],
      comment: json['comment'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'appointmentId': appointmentId,
      'rating': rating,
      'comment': comment,
    };
  }
}
